rakam-postgresql/src/main/java/org/rakam/postgresql/analysis/FastGenericFunnelQueryExecutor.java

```diff
@@ -84,8 +89,9 @@ public QueryExecution query(String project, List<FunnelStep> steps, Optional<Str
             }
 
             insideSelect.add(format("min(case when step = %d then %s end) as ts_event%d", i, checkTableColumn(projectConfig.getTimeColumn()), i));
-            mainSelect.add(format("select %s %d as step, %s, %s from %s where %s between timestamp '%s' and timestamp '%s' and %s",
+            mainSelect.add(format("select %s %s %d as step, %s, %s from %s where %s between timestamp '%s' and timestamp '%s' and %s",
                     dimension.map(v -> checkTableColumn(v) + ", ").orElse(""),
+                    segment.isPresent() ? format(timeStampMapping.get(FunnelTimestampSegments.valueOf(segment.get().replace(" ", "_").toUpperCase())), projectConfig.getTimeColumn())+ " as "  + checkTableColumn(projectConfig.getTimeColumn()+segmentName) + ",":  "",
```

Comment:
Also, why do we use `projectConfig.getTimeColumn()`? The dimension may not always be `_time`.

Reply:
Because of the false assumption I made, that this field segmentation will only be used for time. Will change to accomodate other options.